crabs/detection_tracking/inference_model.py

```diff
@@ -136,50 +153,173 @@ def prep_csv_writer(self):
 
         return csv_writer, csv_file
 
-    def write_bbox_to_csv(self, bbox, frame, frame_name, csv_writer):
+    def evaluate_tracking(
+        self,
+        gt_boxes_list: list,
+        tracked_boxes_list: list,
+        iou_threshold: float,
+    ) -> List[float]:
         """
-        Write bounding box annotation to csv
+        Evaluate tracking performance using the Multi-Object Tracking Accuracy (MOTA) metric.
+
+        Parameters:
+        -----------
+        gt_boxes_list : List[list]
+            List of ground truth bounding boxes for each frame.
+        tracked_boxes_list : List[list]
+            List of tracked bounding boxes for each frame.
+        iou_threshold : float
+            The IoU threshold used to determine matches between ground truth and tracked boxes.
+
+        Returns:
+        --------
+        float:
+            The computed MOTA (Multi-Object Tracking Accuracy) score for the tracking performance.
         """
+        from crabs.detection_tracking.detection_utils import evaluate_mota
 
-        # Bounding box geometry
-        xmin, ymin, xmax, ymax, id = bbox
-        width_box = int(xmax - xmin)
-        height_box = int(ymax - ymin)
-
-        # Add to csv
-        csv_writer.writerow(
-            (
-                frame_name,
-                frame.size,
-                '{{"clip":{}}}'.format("123"),
-                1,
-                0,
-                '{{"name":"rect","x":{},"y":{},"width":{},"height":{}}}'.format(
-                    xmin, ymin, width_box, height_box
-                ),
-                '{{"track":{}}}'.format(id),
+        mota_values = []
+        prev_frame: Union[Optional[List[List[float]]], None] = None
+        for gt_boxes, tracked_boxes in zip(gt_boxes_list, tracked_boxes_list):
+            mota = evaluate_mota(
+                gt_boxes, tracked_boxes, iou_threshold, prev_frame
             )
-        )
+            mota_values.append(mota)
+            # Update previous frame IDs for the next iteration
+            prev_frame = [list(box) for box in tracked_boxes]
+
+        return mota_values
+
+    def get_prediction(self, frame: np.ndarray) -> torch.Tensor:
```

Comment:
```suggestion
    def get_detector_prediction(self, frame: np.ndarray) -> torch.Tensor:
```

Reply:
Can we keep this one, we only have one prediction which is from detector. or we can change to get_detection if it's better for you?

Reply:
I like `get_detection` but not a big deal yeah! My point is that sometimes it's unclear whether prediction is just the bbox or the bbox and the track, but maybe not an issue at this point.